### PR TITLE
Only play the long 'Waking Up' explanation five times

### PR DESCRIPTION
**Pull request by boot-ini**
The default message when Mycroft is going to sleep takes quite a while to complete, which is frustrating when you're trying to get Mycroft to be quiet when you're watching TV or having a conversation. This change tweaks it so after the fifth time Mycroft has explained about waking up he'll just say "Okay" when you tell him to go to sleep.

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -39,8 +39,23 @@ def handle_go_to_sleep(self, message):
...
+            If the user has been told about the waking up process five times
+            already, it sends a shorter message.
         """
-        self.speak_dialog('going.to.sleep', dict(wake_word=self.wake_word))
+        
+        count = self.settings.get('Wake up explanation count')
+        if count is None:
+            count = 1
+        else:
+            count += 1
```
Comment (forslund) [inline __init__.py]: Above can be simplified by setting the default value to 0:
```python
        count = self.settings.get('Wake up explanation count', 0)
        count += 1
```

File: __init__.py [hunk 2]
```diff
@@ -39,8 +39,23 @@ def handle_go_to_sleep(self, message):
...
+        
+        if count <= 5:
+            self.speak_dialog('going.to.sleep', dict(wake_word=self.wake_word))
+            self.settings['Wake up explanation count'] = count
+        else:
+            self.speak_dialog('going.to.sleep.short', dict(wake_word=self.wake_word))
```
Comment (forslund) [inline __init__.py]: This doesn't need the dict(...) part

File: __init__.py [hunk 3]
```diff
@@ -39,8 +39,23 @@ def handle_go_to_sleep(self, message):
         """
             Sends a message to the speech client setting the listener in a
             sleep mode.
+
+            If the user has been told about the waking up process five times
+            already, it sends a shorter message.
         """
-        self.speak_dialog('going.to.sleep', dict(wake_word=self.wake_word))
+        
+        count = self.settings.get('Wake up explanation count')
```
Comment (forslund) [inline __init__.py]: Maybe we could shorten 'Wake up explanation count' value to 'wake up count'

File: dialog/en-us/going.to.sleep.short.dialog [hunk 4]
```diff
@@ -0,0 +1 @@
+Okay.
```
Comment (forslund) [inline dialog/en-us/going.to.sleep.short.dialog]: Maybe add a second line as an alternative with just: `Going to sleep` slightly longer but not by much. (You don't need to translate it to the other languages if that's hard, that will be picked up on translate.mycroft.ai)

**Review by forslund — CHANGES_REQUESTED** [review]
I've added a bunch of comments. Hope you don't think they're unreasonable.

**Comment by boot-ini** [comment]
Thanks for the input, changes are going up now

**Comment by forslund** [comment]
Looks pretty good now thanks!

Before I can include this change we need you to sign our [Contributor License Agreement](https://mycroft.ai/cla). This is a formality to protect yourself, Mycroft AI and our community. This agreement makes it crystal clear that along with your code you are offering a license to use it within the confines of this project. You retain ownership of the code, this is just a license.

This is a one time thing for your future contributions this step won't be needed.

**Comment by boot-ini** [comment]
Done :+1:

**Comment by forslund** [comment]
Thank you, once it's registered I should be able to merge this.

**Review by forslund — APPROVED** [review]

**Comment by forslund** [comment]
Hi @StefanBCN, I've merged this into the 19.02 branch via #27. Thanks for the contribution!

**Pull request closed without merging**